packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * WordPress dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { comment as commentIcon } from '@wordpress/icons';
+
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import PluginSidebar from '../plugin-sidebar';
+import { collabSidebarName } from './constants';
+import { Comments } from './comments';
+import { AddComment } from './add-comment';
+
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
+
+const modifyBlockCommentAttributes = ( settings, name ) => {
+	if ( name && name?.includes( 'core/' ) ) {
+		if (
+			undefined === settings.attributes.blockCommentId ||
+			null === settings.attributes.blockCommentId
+		) {
```

Comment:
A nested `if` is not great for readability. Can we put the condition in the above if?

Also, this may likely be simplified to just `! settings.attributes.blockCommentId`:

```suggestion
		if ( ! settings.attributes.blockCommentId ) {
```